code/datums/traits/negative.dm

```diff
@@ -100,94 +100,339 @@
 	if(DT_PROB(0.05, delta_time))
 		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "depression", /datum/mood_event/depression)
 
+
 /datum/quirk/family_heirloom
 	name = "Family Heirloom"
 	desc = "You are the current owner of an heirloom, passed down for generations. You have to keep it safe!"
-	value = -1
+	value = -2
 	mood_quirk = TRUE
 	var/obj/item/heirloom
 	var/where
 
 /datum/quirk/family_heirloom/on_spawn()
 	var/mob/living/carbon/human/H = quirk_holder
 	var/obj/item/heirloom_type
+	var/list/heirloom_table = list()
+	var/civ = FALSE
+	var/med = FALSE
+	var/sec = FALSE
+	var/sci = FALSE
+	var/engi = FALSE
+	var/supp = FALSE
+	var/comm = FALSE
+
+	//------------------------------//
+	// Adding items from this point //
+	//------------------------------//
+
+	// 1. Species specific table
+	// '|| prob(1)' means you can get other race's heirloom at low chance.
+	//Human
+	if(is_species(H, /datum/species/human) || prob(1))
+		heirloom_table.Add(/obj/item/clothing/head/kitty) //you had terrible parents.
+	//Felinid
+	if(is_species(H, /datum/species/human/felinid) || prob(1))
+		heirloom_table.Add(/obj/item/clothing/head/kitty)
+	//Lizard
+	if(is_species(H, /datum/species/lizard) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/lizardplushie,
+						   /obj/item/clothing/under/costume/gladiator)
+	//Oozeling
+	if(is_species(H, /datum/species/oozeling) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/slimeplushie)
+	//Moth
+	if(is_species(H, /datum/species/moth) || prob(1))
+		heirloom_table.Add(/obj/item/flashlight/lantern/heirloom_moth,
+						   /obj/item/toy/plush/moth)
+	//Apid
+	if(is_species(H, /datum/species/apid) || prob(1))
+		heirloom_table.Add(/obj/item/toy/plush/beeplushie)
+	//Plamaman
+	if(is_species(H, /datum/species/plasmaman) || prob(1))
+		heirloom_table.Add(/obj/item/coin/plasma)
+	//Ethereal
+	if(is_species(H, /datum/species/ethereal) || prob(1))
+		heirloom_table.Add(/obj/item/coin/plasma) //I am not sure what to give them
+	//IPC
+	if(is_species(H, /datum/species/ipc) || prob(1))
+		heirloom_table.Add(/obj/item/disk/data)
```

Comment:
those checks are kinda stupid and if anything they should use else if
and instead of .Add just use +=

Reply:
I used 'else if' at first, but changed my mind to check 1% for every check.

Reply:
if I say why I shouldn't use 'else if' to each check, let's say you're a human, then you just get a human heirloom at first check and you won't get other races' heirloom.

Reply:
You can't be multiple species and yes switch is better here

Reply:
+= doesn't work on paths spent an hour yesterday finding a method to find out how to and the method ended up being Add